### PR TITLE
enhancement/simulation

### PR DESCRIPTION
**Pull request by decanus**
- [x] configuration for offline percentage
- [x] configuration for node count
- [x] configuration for messages second interval
- [x] configuration for how many nodes are sending messages

### CODE REVIEW

File: simulation/simulation.go [hunk 1]
```diff
@@ -33,54 +44,54 @@ func (t *Transport) Send(group mvds.GroupID, sender mvds.PeerId, peer mvds.PeerI
...
 
-	na.Share(group, nb.ID)
-	na.Share(group, nc.ID)
+	group := groupId()
+	// @todo add multiple groups, only one or so nodes in every group so there is overlap
+	// @todo maybe dms?
 
-	nb.Share(group, na.ID)
-	nb.Share(group, nc.ID)
+	// @todo allow for not all nodes to be peered and sharing to test how that looks
```
Comment (oskarth) [inline simulation/simulation.go]: ah was gonna comment about this, yeah leaving for later seems fine

File: simulation/simulation.go [hunk 2]
```diff
@@ -74,7 +74,11 @@ func main() {
 	// @todo add multiple groups, only one or so nodes in every group so there is overlap
 	// @todo maybe dms?
 
-	// @todo allow for not all nodes to be peered and sharing to test how that looks
+	// @todo allow for selecting how many nodes will be paired, this works as follows:
```
Comment (oskarth) [inline simulation/simulation.go]: shared

**Review by oskarth — APPROVED** [review]

File: simulation/simulation.go [hunk 3]
```diff
@@ -110,9 +121,12 @@ func peerId() mvds.PeerId {
 	return mvds.PeerId(key.PublicKey)
 }
 
-func groupId(n string) mvds.GroupID {
-	bytes := []byte(n)
+func groupId() mvds.GroupID {
+	bytes := make([]byte, 32)
+	rand.Read(bytes)
```
Comment (oskarth) [inline simulation/simulation.go:160]: What's this for?

Comment (decanus) [reply simulation/simulation.go:160]: @oskarth creates random group ids

**Pull request merged**